src/adapter/graphql/src/mutations/account_mut.rs

```diff
@@ -57,6 +70,81 @@ impl AccountMut {
         }
     }
 
+    /// Create a new account
+    async fn create_account(
+        &self,
+        ctx: &Context<'_>,
+        account_name: String,
+        email: Option<String>,
+    ) -> Result<CreateAccountResult> {
+        ensure_account_can_provision_accounts(ctx, &self.account.id).await?;
+
+        let Ok(account_name) = odf::AccountName::from_str(&account_name) else {
+            return Ok(CreateAccountResult::InvalidAccountName(
+                AccountNameInvalid::default(),
+            ));
+        };
+
+        let new_email = email.unwrap_or(
+            if let Some(parent_account_parts) = self.account.email.to_string().split_once('@') {
+                format!(
+                    "{}+{}@{}",
+                    parent_account_parts.0, account_name, parent_account_parts.1
+                )
+            } else {
+                format!("{account_name}@example.com",)
+            },
+        );
+
+        let Ok(new_email) = Email::parse(&new_email) else {
+            return Ok(CreateAccountResult::InvalidEmail(
+                CreateAccountEmailInvalid::default(),
+            ));
+        };
+
+        let random_password =
+            random_strings::get_random_string(None, 10, &AllowedSymbols::AsciiSymbols);
+
+        let account_service = from_catalog_n!(ctx, dyn AccountService);
+        match account_service
+            .create_account(&account_name, new_email, random_password, &self.account.id)
+            .await
+        {
+            Ok(created_account) => Ok(CreateAccountResult::Success(CreateAccountSuccess {
+                account: AccountView::from_account(created_account),
+            })),
+            Err(CreateAccountError::Duplicate(err)) => Ok(
+                CreateAccountResult::NonUniqueAccountField(AccountFieldNonUnique {
+                    field: err.account_field.to_string(),
+                }),
+            ),
+            Err(CreateAccountError::Internal(e)) => Err(e.into()),
+        }
+    }
+
+    /// Reset password for selected account. Allowed only for admin users
+    #[graphql(guard = "AdminGuard::new()")]
+    async fn modify_password(
```

Comment:
Why the permission check is different from `create_account`? I don't think we are going to give Molecule an admin account for the whole system.

Reply:
Who wil be modifying the passwords, us for all customers?